[SEDONA-197] Added ST_ZMax and ST_ZMin to Apache Sedona

Varsha-R opened the pull request:
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/community/develop/)


## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-197. The PR name follows the format `[SEDONA-XXX] my subject`.


## What changes were proposed in this PR?
Added ST_ZMax and ST_ZMin functions to SQL

## How was this patch tested?
Through unit tests written in Scala

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation.

jiayuasu commented on docs/api/sql/Function.md:
```diff
@@ -1445,3 +1445,33 @@ SELECT ST_Z(ST_POINT(0.0 25.0 11.0))
+
+Format: `ST_ZMax(geom: geometry)`
+
+Since: `v1.2.1`
```
the version number should be 1.3.1

Varsha-R replied:
Updated it

jiayuasu reviewed (changes requested):
Please add the following:

1. DataFrameAPI for Spark, with tests: https://github.com/apache/incubator-sedona/blob/master/sql/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala
2. Corresponding function in Flink, with tests: https://github.com/apache/incubator-sedona/blob/master/flink/src/main/java/org/apache/sedona/flink/Catalog.java#L21
3.

Varsha-R commented:
> Please add the following:
> 
> 1. DataFrameAPI for Spark, with tests: https://github.com/apache/incubator-sedona/blob/master/sql/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala
> 2. Corresponding function in Flink, with tests: https://github.com/apache/incubator-sedona/blob/master/flink/src/main/java/org/apache/sedona/flink/Catalog.java#L21

Addressed these comments in [this commit](https://github.com/apache/incubator-sedona/pull/717/commits/b9fffd6ea2830350128141c548372613b658fe00#diff-42591a191598418405b17af4a87d795eb556c69b5233d699e4c106230c0e99ef)

jiayuasu reviewed (changes requested):
Looks good to me. One last thing: please add the DataFrame style API to Python as well -> https://github.com/apache/incubator-sedona/blob/master/python/sedona/sql/st_functions.py
